### PR TITLE
vli:add VL822 device+share-spi-pd, in testing

### PR DESCRIPTION
**Pull request by memily**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

Our client requests to be added onto fwupd, their Share SPI device is VL822+VL103. VL822 is a new type of hardware, it is an upgrade from VL820 with some minor adjustments that I added in this PR. Everything else related to how VL822 should be updated is the same as VL817. 

As I started testing I ran into the same issue as Lenovo TR Gen 2 where fwupd finds the Hub's VID/PID version for PD but does not erase/write to the Share SPI PD. Hub can be upgrade/downgrade with success. I forgot how this was originally resolved and I was hoping you, perhaps, remember! Please take a look at your convenience, thanks.

### CODE REVIEW

File: plugins/vli/fu-vli-usbhub-firmware.c [hunk 1]
```diff
@@ -174,16 +174,33 @@ fu_vli_usbhub_firmware_parse (FuFirmware *firmware,
 		self->device_kind = FU_VLI_DEVICE_KIND_VL211;
 		break;
 	case 0x0518:
-		/* VL820 == VT3518 */
-		if (!fu_common_read_uint8_safe (buf, bufsz, 0xf000, &tmp, error)) {
-			g_prefix_error (error, "failed to get Q7/Q8 difference: ");
+	{
```
Comment (hughsie) [inline plugins/vli/fu-vli-usbhub-firmware.c:177]: I don't think you need this brace

**Comment by hughsie** [comment]
Also,. CI failure, should be easy to fix: 

    Bail out! FuPluginVliUsbhub:ERROR:../plugins/vli/fu-self-test.c:20:fu_test_common_device_kind_func: assertion failed (fu_vli_common_device_kind_from_string (tmp) == i): (0 == 2082)

File: plugins/vli/fu-vli-usbhub-firmware.c (hunk 1 above)
Comment (memily) [reply plugins/vli/fu-vli-usbhub-firmware.c:177]: Oops, will do, is this also the CI failure?

Comment (hughsie) [reply plugins/vli/fu-vli-usbhub-firmware.c:177]: No, this is me being a pedant :)

Comment (memily) [reply plugins/vli/fu-vli-usbhub-firmware.c:177]: En garde! The case 0518 block trips up with compiler without those brackets on our machines :/ . Where's the Clang code formatter?

Comment (hughsie) [reply plugins/vli/fu-vli-usbhub-firmware.c:177]: > trips up with compiler

That doesn't sound awesome. What compile failure you getting? It works with GCC here.

**Comment by memily** [comment]
Minor update: our FAE department has contacted the client, and they should start the process of sending you a device for testing.

**Comment by hughsie** [comment]
> they should start the process of sending you a device

Agreed, I send my address just this morning.

**Comment by memily** [comment]
Should we upload test fw to lvfs or to separate git repo?

**Comment by hughsie** [comment]
> Should we upload test fw to lvfs

Please -- then we can be sure there is legal permission to use it in this way.

**Comment by memily** [comment]
Please add BizLink VID 0x06C4 access to the vli account on lvfs. Thanks!

**Comment by hughsie** [comment]
> Please add BizLink VID 0x06C4

Done!

**Comment by memily** [comment]
> Done! 

Our hero! We send our eternal gratitude.

**Comment by memily** [comment]
I forgot to fix the brackets in plugins/vli/fu-vli-usbhub-firmware.c line 177, but how should I fix the CI failures?

**Comment by hughsie** [comment]
You need to add the missing string mapping of `FU_VLI_DEVICE_KIND_VL822` to `fu_vli_common_device_kind_from_string()` in `plugins/vli/fu-vli-common.c`

**Pull request merged**